dll/win32/shell32/utils.cpp

```diff
@@ -99,3 +99,135 @@ SHFindComputer(LPCITEMIDLIST pidlRoot, LPCITEMIDLIST pidlSavedSearch)
 
     return SUCCEEDED(hr);
 }
+
+static VOID Int64ToStr(LONGLONG llValue, LPWSTR pszValue)
+{
+    WCHAR szBuff[40];
+    INT ich = 0;
+#if (WINVER >= 0x0600)
+    BOOL bMinus = (llValue < 0);
+
+    if (bMinus)
+        llValue = -llValue;
+#endif
+
+    do
+    {
+        szBuff[ich++] = (WCHAR)(L'0' + (llValue % 10));
+        llValue /= 10;
+    } while (llValue != 0);
+
+#if (WINVER >= 0x0600)
+    if (bMinus)
+        szBuff[ich++] = '-';
+#endif
+
+    do
+    {
+        --ich;
+        *pszValue++ = szBuff[ich];
+    } while (ich > 0);
+
+    *pszValue = UNICODE_NULL;
+}
+
+#define MAX_NUMSEP 6
+
+static VOID
+Int64GetNumFormat(NUMBERFMTW *pDest, const NUMBERFMTW *pSrc, DWORD dwNumberFlags,
+                  LPWSTR pszDecimal, LPWSTR pszThousand)
+{
+    WCHAR szBuff[20];
+
+    if (pSrc)
+        *pDest = *pSrc;
+    else
+        dwNumberFlags = 0;
+
+    if (!(dwNumberFlags & NUMFLAG_NUMDIGITS))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_IDIGITS, szBuff, _countof(szBuff));
+        pDest->NumDigits = StrToIntW(szBuff);
+    }
+
+    if (!(dwNumberFlags & NUMFLAG_LEADINGZERO))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_ILZERO, szBuff, _countof(szBuff));
+        pDest->LeadingZero = StrToIntW(szBuff);
+    }
+
+    if (!(dwNumberFlags & NUMFLAG_GROUPING))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_SGROUPING, szBuff, _countof(szBuff));
+        pDest->Grouping = StrToIntW(szBuff);
+    }
+
+    if (!(dwNumberFlags & NUMFLAG_DECIMALSEP))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_SDECIMAL, pszDecimal, MAX_NUMSEP);
+        pDest->lpDecimalSep = pszDecimal;
+    }
+
+    if (!(dwNumberFlags & NUMFLAG_THOUSANDSEP))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_STHOUSAND, pszThousand, MAX_NUMSEP);
+        pDest->lpThousandSep = pszThousand;
+    }
+
+    if (!(dwNumberFlags & NUMFLAG_NEGATIVEORDER))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_INEGNUMBER, szBuff, _countof(szBuff));
+        pDest->NegativeOrder = StrToIntW(szBuff);
+    }
+}
+
+/*************************************************************************
+ *  Int64ToString [SHELL32.209]
+ */
+EXTERN_C
+INT WINAPI
+Int64ToString(
+    _In_ LONGLONG llValue,
+    _Out_ LPWSTR pszOut,
+    _In_ UINT cchOut,
+    _In_ BOOL bUseFormat,
+    _In_ const NUMBERFMTW *pNumberFormat,
+    _In_ DWORD dwNumberFlags)
+{
+    INT ret;
+    NUMBERFMTW NumFormat;
+    WCHAR szValue[80], szDecimalSep[MAX_NUMSEP], szThousandSep[MAX_NUMSEP];
+
+    Int64ToStr(llValue, szValue);
```

Comment:
It would be better if the internal Int64ToStr function could take the size of the szValue buffer, in order to make it clear it can handle a potential ill-sized buffer.

Reply:
Fixed.